File: a135(language_detection)/Python/language_detection.py
from sys import stdin, stdout


HELLO_OF_COUNTRIES = {
    'HELLO': 'ENGLISH',
    'HOLA': 'SPANISH',
    'HALLO': 'GERMAN',
    'BONJOUR': 'FRENCH',
    'CIAO': 'ITALIAN',
    'ZDRAVSTVUJTE': 'RUSSIAN',
}


for i, line in enumerate(stdin):
    line = line.strip()
    if line == '#':
        break
        
    stdout.write(f"Case {i+1}: {HELLO_OF_COUNTRIES.get(line, 'UNKNOWN')}\n")

File: a135(language_detection)/C++/language_detection.cpp
#include <iostream>
#include <string>
#include <unordered_map>


void OptimizeIo() {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(0);
}


void PrintCountryOfCase(const std::string& country, const int case_no) {
  std::cout << "Case " << case_no << ": " << country << '\n';
}


int main(int argc, char const *argv[]) {
  OptimizeIo();

  const std::unordered_map<std::string, std::string> HELLO_OF_COUNTRIES = {
    {"HELLO", "ENGLISH"},
    {"HOLA", "SPANISH"},
    {"HALLO", "GERMAN"},
    {"BONJOUR", "FRENCH"},
    {"CIAO", "ITALIAN"},
    {"ZDRAVSTVUJTE", "RUSSIAN"},
  };

  std::string line{};
  for (size_t i = 1;
       std::cin >> line && line != "#";
       ++i) {
    const auto country_iter = HELLO_OF_COUNTRIES.find(line);

    PrintCountryOfCase(
      country_iter == HELLO_OF_COUNTRIES.cend()
          ? "UNKNOWN" : country_iter->second,
      i
    );
  }

  return 0;
}
